src/auth/index.js

```diff
@@ -214,27 +214,25 @@ export default class Auth {
   }
 
   setAuthType(authType) {
-    return new Promise((resolve) =>
-      resolve(this.storage.set(authCommon.USER_LOGGED_IN_PT_KEY, authType))
-    );
+    return this.storage.set(authCommon.USER_LOGGED_IN_PT_KEY, authType);
   }
 
-  set(json, authType) {
+  set(json, authType = '') {
     if (!json) {
       return;
     }
 
     let newUserAuth = {};
-    return new Promise((resolve) => {
-      if (!authType) { resolve(); }
-      else { resolve(this.storage.set(authCommon.USER_LOGGED_IN_PT_KEY, authType)); }
-    }).then(() => {
+
+    return (authType ? 
+      this.storage.set(authCommon.USER_LOGGED_IN_PT_KEY, authType) : 
+      new Promise()
```

Comment:
I think this needs to be `Promise.resolve()`, rather than `new Promise()`.

Reply:
`new Promise()` has been tested and works. But if you'd prefer `Promise.resolve()`, I can alter it.

Reply:
Are you sure? I think you're required to pass a function to the `Promise` constructor otherwise there's no way for it to hook into the following `.then`/`.catch`.
e.g. if i do that I get this:
```
> (new Promise()).then(()=>{console.log("hi")})
TypeError: Promise resolver undefined is not a function
    at Promise (<anonymous>)
```

if i use `Promise.resolve` though i get:
```> (Promise.resolve()).then(()=>{console.log("hi")})
Promise {
  <pending>,
  domain:
   Domain {
     domain: null,
     _events: { error: [Function: debugDomainError] },
     _eventsCount: 1,
     _maxListeners: undefined,
     members: [] } }
> hi
```

Reply:
Fixed